andela/ah-codeblooded-frontend
# #160577594 Enable users to see their article stats

**What does this PR do?**

This PR adds functionality that allows users to see how their articles are performing on the platform on a single dashboard.

**Description of Task to be completed?**

The article stats data gives the user data about the performance of their articles. Only the owner is able to see this data. This means that one has to be logged in to view their reading stats.

**How should this be manually tested?**

This feature can be manually tested by opening this PR's [review app](https://ah-cd-frontend-staging-pr-35.herokuapp.com/).
1. Log in to the app.
2. Click on the `Stats` link on the NavBar. You should be taken to a page that displays your article stats.

**Any background context you want to provide?**

When a backend PR adding like count, dislike count and rating stats is merged, they will also be added.

**What are the relevant pivotal tracker stories?**

[#160577594](https://www.pivotaltracker.com/story/show/160577594)

**Screenshots**
NavBar link
<img width="914" alt="screen shot 2018-11-27 at 23 33 15" src="https://user-images.githubusercontent.com/8180548/49109872-f0964500-f29c-11e8-98c8-2dbd11b68bb1.png">

Loading stats
<img width="854" alt="screen shot 2018-11-23 at 11 56 31" src="https://user-images.githubusercontent.com/8180548/48935068-60ac7000-ef17-11e8-8e6d-0a95db0e763b.png">

No stats
<img width="1093" alt="screen shot 2018-11-23 at 08 38 53" src="https://user-images.githubusercontent.com/8180548/48929593-50869780-eefb-11e8-941c-95292e7cbde4.png">

Stats on display
<img width="823" alt="screen shot 2018-11-23 at 11 53 09" src="https://user-images.githubusercontent.com/8180548/48935105-7de13e80-ef17-11e8-812d-6b7055fa94e2.png">

Refreshing stats - notice the preloader on the NavBar
<img width="802" alt="screen shot 2018-11-27 at 23 35 56" src="https://user-images.githubusercontent.com/8180548/49109994-466aed00-f29d-11e8-9eac-8befa7160cda.png">
